recipe_scrapers/thecookingguy.py

```diff
@@ -0,0 +1,100 @@
+# mypy: allow-untyped-defs
+from ._abstract import AbstractScraper
+from ._exceptions import ElementNotFoundInHtml
+from ._grouping_utils import IngredientGroup
+from ._utils import get_yields, normalize_string
+
+
+class TheCookingGuy(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "thecookingguy.com"
+
+    def author(self):
+        return self.schema.author()
+
+    def title(self):
+        return self.schema.title()
+
+    def total_time(self):
+        return None
```

Comment:
Does this mean that no recipes from thecookingguy.com seem to have total times listed?  (I can believe it, but just checking that that's the intended behaviour here)

Reply:
I think so yes, they have times within the instructions but not a single field for total time. See here for an example: https://www.thecookingguy.com/recipes/chili-cumin-lamb

Reply:
This isn't something I've suggested before, so it would be a bit novel, but: what do you think about emitting a [Python warning](https://docs.python.org/3/library/warnings.html) -- perhaps even one that would be ignored by default -- before returning the empty (not-known-to-be-available) value?

Reply:
At the risk of seeming spammy or self-absorbed, let me provide some more context here; I generally have a few types of end-user in mind for this library:

1. Enthusiasts/hobbyists who want to find an accessible way to read a small number of recipes from the web, or who may be learning some fun/straightforward Python code, and who may be running the code partly/entirely from the Python interpreter.
2. Slightly more advanced users -- let's call them tech parents of the 2020s -- who are able to create a spreadsheet and run some Python code, and may be doing things like importing a group of recipes that the family likes, or some candidate options, into a spreadsheet for filtering/reference purposes.
3. Larger, probably automated sites that are crawling the web itself, or perhaps using CommonCrawl, to assess larger numbers of recipes for academic/commercial/research purposes.

As a maintainer and developer, I generally want all three of these groups to have a good developer/user experience, but also to encourage them to behave well, and to notify them about potential problems in their usage patterns and the data.

In particular, for this field method that we always expect to return `None` -- so it's _kinda_ pointless to invoke -- I'd like use-cases (1) and (3) to get some kind of indication -- whether it's in their individual console or their haystack of logs -- that for a given website, there's no information available from a particular field.  For use-case (2), it's _probably_ fine to pass by silently -- ideally we'd perhaps autopopulate a comment or note in the spreadsheet to indicate that an individual row isn't available from that source (phrased in a clearer way than I just attempted), but doing so does incur a risk of distracting the person from their objectives.

I'm also likely to be biased because my own use-case is (3) as a small-business owner that crawls the web -- but to a certain extent that also makes me want to boost the other use-cases because I want the project to remain healthy (and I think that that depends on the other two cases).

Reply:
I think we can do that, could we get some more details on what that looks like? Would it just be calling something like 

```
warnings.filterwarnings('ignore', message='not allowed') 
warnings.warn("Not known to be available")
```
before returning?

Reply:
Or maybe we could leave out `warnings.filterwarnings('ignore', message='Not known to be available')`, which would hide the warning from people running it from the interpreter.

Reply:
We've never worked with the warnings package, so a little bit extra guidance on how to do that would be appreciated!

Reply:
That's ok, it's fairly new to me too, so it's a learning experience all round :)

I think we should confirm whether calling `ignorewarnings` in the example you gave would override command-line / environment warning settings; if so, that's a potential problem (because we do want the warning to be emitted sometimes).

None of the default-ignored Python warning categories seemed to match this situation as far as I can tell.

In terms of a message format, I think we should try to write the error so that it could apply to other fields in future:

```
warnings.warn(f"Hm.  Apparently the {website} doesn't provide {field} values?  If you know that's untrue for some recipe, let us know at {bugreport-url}")
```

Reply:
Oops, `filterwarnings` I mean (not `ignorewarnings`).

Reply:
So with filterwarnings I get no warning from interpreter:
```
>>> from recipe_scrapers import scrape_me
>>> scraper = scrape_me("https://www.thecookingguy.com/recipes/chili-cumin-lamb")
>>> scraper.total_time()
```
Without it I get
```
>>> from recipe_scrapers import scrape_me
>>> scraper = scrape_me("https://www.thecookingguy.com/recipes/chili-cumin-lamb")
>>> scraper.total_time()
/home/rmdluo/recipe-scrapers/recipe_scrapers/thecookingguy.py:21: UserWarning: Not known to be available
  warnings.warn("Not known to be available")
```

And then it looks like for changing when warnings are displayed, you can add some code that filters warnings based on whether a certain flag was used or not: https://docs.python.org/3/library/warnings.html#default-warning-filter -- not sure if that is something we would be doing or something that should be happening on a higher level.

Reply:
Ok, thanks.  Yep, we don't want to skip the warnings entirely (your first sample code).

I think what we'd want is a warning that appears when the `-Walways` (always warn) flag is set on the commandline, but that does _not_ appear with the default (`-Wdefault`).

In essence it feels something like a `PotentiallyPointlessInvocationWarning` (although I'm certain there are better names than that).

Reply:
I have it set up now so that if you do
```
python -Walways
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from recipe_scrapers import scrape_me
>>> scraper = scrape_me("https://www.thecookingguy.com/recipes/chili-cumin-lamb")
>>> scraper.total_time()
/home/rmdluo/recipe-scrapers/recipe_scrapers/thecookingguy.py:24: UserWarning: Not known to be available
  warnings.warn("Not known to be available")
```
the warning appears, but if you do
```
python
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from recipe_scrapers import scrape_me
>>> scraper = scrape_me("https://www.thecookingguy.com/recipes/chili-cumin-lamb")
>>> scraper.total_time()
```
you get no warning. Does this seem about right?

Reply:
That's the behaviour I had in mind, yep - was it possible to achieve that without quirky/unusual code?  (if the code to achieve it seem complicated, then I think we might want to go with a simpler approach instead.  in any case though, thank you for investigating this)

Reply:
I went ahead and made the warning message more descriptive as well

Reply:
Looks like it also displays the warning when calling `python -m unittest -k thecookingguy`, not sure if that is desirable

Reply:
In some ways that is quite useful I think, because it provides us with some context about how many scraper fields are affected by this.  However: that output can seem a bit noisy when compared to the fairly clean default output of `unittest`.

Again I'm afraid I don't have neat simple answers about how to handle this yet.  We could add some filtering in our continuous integration to filter the message there -- or collect them so that they're somehow aggregated in the test report output -- and/or do the same for local development.